arangod/Cluster/ServerState.h

```diff
@@ -292,6 +301,10 @@ class ServerState {
   /// @brief the server's short id, can be set just once
   std::atomic<uint32_t> _shortId;
 
+  /// @brief the server's rebootId, can be set just once and
+  ///        is usually read while integrating into a cluster
+  std::atomic<uint64_t> _rebootId;
```

Comment:
I don't think this has to be an atomic.

Reply:
about everything else in this class is protected at some point; would it being atomic incur any additional problems? would it *not* being atomic be an advantage?

(I do not know how ServerState is accessed in arangod and which bits of it need to be protected against concurrent access: indeed @mchacki thought that we do not need to store the rebootId in ServerState at all, @neunhoef seemed to suggest we should. The reason why I read it back is that I would like to assert that rebootId is 1 if I generated a new uuid earlier, but we do not need to store the reebotId in the server state if we do not use it at any time)

Reply:
It's not obvious to me that we need it here, either. But maybe its useful.

On the atomic: Operations on atomics are slower. This is probably of no consequence here.

But if its an atomic, it looks like there are concurring reads/writes on it. Which should definitely not happen as I see it. But I would stumble across that and check my assumptions as a reader if it's an atomic, when I think it doesn't have to be.